sas9.4-viya/azuredeploy.json

```diff
@@ -0,0 +1,3683 @@
+        "key_vault_secretname_pvtkey": "ansible-pvtkey",
+        "key_vault_secretname_sasext": "sasextpw",
+        "key_vault_secretname_sasinst": "sasintpw",
+        "key_vault_secretname_stgacc": "stgacckey",
+        "keysPermissions": [
+            "get",
+            "list",
+            "import"
+        ],
+        "linux_extension_template": "[concat(parameters('_artifactsLocation'),'nestedtemplates/vm_linux_extension.json', variables('_artifactsLocationSasToken'))]",
+        "meta_config_sh": "[concat(parameters('_artifactsLocation'),'scripts/meta_config.sh', variables('_artifactsLocationSasToken'))]",
+        "meta_install_sh": "[concat(parameters('_artifactsLocation'),'scripts/meta_install.sh', variables('_artifactsLocationSasToken'))]",
+        "meta_nw_interface": "[concat(parameters('SAS Application Name'),'_',variables('meta_vm_name'),'_nic_',variables('resourceGroupUniqueString'))]",
+        "meta_nw_sg": "[concat(parameters('SAS Application Name'),'_',variables('meta_vm_name'),'_nsg_',variables('resourceGroupUniqueString'))]",
+        "meta_vm_name": "meta",
+        "microservices_nw_interface": "[concat(parameters('SAS Application Name'),'_',variables('microservices_vm_name'),'_nic_',variables('resourceGroupUniqueString'))]",
+        "microservices_nw_sg": "[concat(parameters('SAS Application Name'),'_',variables('microservices_vm_name'),'_nsg_',variables('resourceGroupUniqueString'))]",
+        "microservices_vm_name": "microservices",
+        "mid_config_sh": "[concat(parameters('_artifactsLocation'),'scripts/mid_config.sh', variables('_artifactsLocationSasToken'))]",
+        "mid_install_sh": "[concat(parameters('_artifactsLocation'),'scripts/mid_install.sh', variables('_artifactsLocationSasToken'))]",
+        "mid_nw_interface": "[concat(parameters('SAS Application Name'),'_',variables('mid_vm_name'),'_nic_',variables('resourceGroupUniqueString'))]",
+        "mid_nw_sg": "[concat(parameters('SAS Application Name'),'_',variables('mid_vm_name'),'_nsg_',variables('resourceGroupUniqueString'))]",
+        "mid_vm_name": "mid",
+        "primary_user_name": "vmuser",
+        "pub_sub_nw_sg": "[concat(parameters('SAS Application Name'), '_subnt_pub1_nsg_',variables('resourceGroupUniqueString'))]",
+        "pvt_sub_nw_sg": "[concat(parameters('SAS Application Name'), '_subnt_pvt1_nsg_',variables('resourceGroupUniqueString'))]",
+        "rdp_nw_interface": "[concat(parameters('SAS Application Name'),'_',variables('rdp_vm_name'),'_nic_',variables('resourceGroupUniqueString'))]",
+        "rdp_nw_sg": "[concat(parameters('SAS Application Name'),'_',variables('rdp_vm_name'),'_nsg_',variables('resourceGroupUniqueString'))]",
+        "rdp_os_version": "2012-R2-Datacenter",
+        "rdp_vm_name": "rdp",
+        "reader_role": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')]",

```

Comment:
remove concat() use subscriptionResourceId()

Reply:
modified and used subscriptionResourceId()